sdk/src/errors.rs

```diff
@@ -321,3 +322,82 @@ pub(crate) fn status_to_error(status: tonic::Status) -> MomentoError {
+                "Request rate limit exceeded for this account".to_string()
+            }
+            LimitExceededMessageWrapper::ThroughputRate => {
+                "Bandwidth limit exceeded for this account".to_string()
+            }
+            LimitExceededMessageWrapper::RequestSize => {
+                "Request size limit exceeded for this account".to_string()
+            }
+            LimitExceededMessageWrapper::ItemSize => {
+                "Item size limit exceeded for this account".to_string()
+            }
+            LimitExceededMessageWrapper::ElementSize => {
+                "Element size limit exceeded for this account".to_string()
+            }
+            LimitExceededMessageWrapper::Unknown => "Limit exceeded for this account".to_string(),
+        }
+    }
+}
+
+fn determine_limit_exceeded_message_wrapper(metadata: &MetadataMap, message: &str) -> String {
+    // If provided, we use the `err` metadata value to determine the most
+    // appropriate error message to return.
+    if let Some(err_cause) = metadata.get("err") {
+        if let Ok(err_str) = err_cause.to_str() {
+            return match err_str {
+                "topic_subscriptions_limit_exceeded" => {
+                    LimitExceededMessageWrapper::TopicSubscriptions.value()
+                }
+                "operations_rate_limit_exceeded" => {
+                    LimitExceededMessageWrapper::OperationsRate.value()
+                }
+                "throughput_rate_limit_exceeded" => {
+                    LimitExceededMessageWrapper::ThroughputRate.value()
+                }
+                "request_size_limit_exceeded" => LimitExceededMessageWrapper::RequestSize.value(),
+                "item_size_limit_exceeded" => LimitExceededMessageWrapper::ItemSize.value(),
+                "element_size_limit_exceeded" => LimitExceededMessageWrapper::ElementSize.value(),
+                _ => LimitExceededMessageWrapper::Unknown.value(),
+            };
+        }
+    }
+
+    // If `err` metadata is unavailable, try to use the error details field
+    // to return an appropriate error message.
+    let lower_cased_message = message.to_lowercase();
+    if lower_cased_message.contains("subscribers") {
+        LimitExceededMessageWrapper::TopicSubscriptions.value()
+    } else if lower_cased_message.contains("operations") {
+        LimitExceededMessageWrapper::OperationsRate.value()
+    } else if lower_cased_message.contains("throughput") {
+        LimitExceededMessageWrapper::ThroughputRate.value()
+    } else if lower_cased_message.contains("request limit") {
+        LimitExceededMessageWrapper::RequestSize.value()
+    } else if lower_cased_message.contains("item size") {
+        LimitExceededMessageWrapper::ItemSize.value()
+    } else if lower_cased_message.contains("element size") {
+        LimitExceededMessageWrapper::ElementSize.value()
+    } else {
+        LimitExceededMessageWrapper::Unknown.value()
+    }
```

Comment:
Could also simplify this by 
```suggestion
    let wrapper = if lower_cased_message.contains("subscribers") {
        LimitExceededMessageWrapper::TopicSubscriptions
    } else if lower_cased_message.contains("operations") {
        LimitExceededMessageWrapper::OperationsRate
    } else if lower_cased_message.contains("throughput") {
        LimitExceededMessageWrapper::ThroughputRate.value()
    } else if lower_cased_message.contains("request limit") {
        LimitExceededMessageWrapper::RequestSize
    } else if lower_cased_message.contains("item size") {
        LimitExceededMessageWrapper::ItemSize
    } else if lower_cased_message.contains("element size") {
        LimitExceededMessageWrapper::ElementSize
    } else {
        LimitExceededMessageWrapper::Unknown
    }
    wrapper.value()
```